app/dashboard/sql/persona.py

```diff
@@ -0,0 +1,322 @@
+        union
+        select profile_id, status::text from not_active_bounty_hunters_completion
+    ) a
+    on
+        c.profile_id = a.profile_id
+    left join (
+        select profile_id, status::text from active_bounty_hunters_action
+        union
+        select profile_id, status::text from active_bounty_hunters_completion
+    ) b
+    on
+        c.profile_id = b.profile_id
+    left join (
+        select profile_id, status::text from active_funders_completion
+        union
+        select profile_id, status::text from active_funders_action
+    ) f
+    on
+        c.profile_id = f.profile_id
+    left join (
+        select profile_id, status::text from not_active_funders_completion
+        union
+        select profile_id, status::text from not_active_funders_action
+    ) g
+    on
+        c.profile_id = g.profile_id
+    where
+        c.email is not null
+    order by
+        c.email
+),
+funder_restriction as (
+    select distinct
+        db.bounty_owner_email
+    from
+        dashboard_bounty db
+    where
+db.network = %s
+        and
+            db.current_bounty = True
+        and
+            db.idx_status in ('open', 'started', 'submitted', 'done', 'expired', 'cancelled')
+        and
+            db.bounty_owner_github_username not in (select au.username from auth_user au where au.is_staff is True and au.username not in ('consensys', 'superuser', 'gitcoinbot') and au.email != '')
+        and
+            db.bounty_owner_github_username not in ('willsputra', 'gitcointestuser2', '', 'austintgriffith')
+),
+result_raw as (
+    select
+        a.profile_id,
+        a.email,
+        case when a.status_bounty_hunter is not null then 'bounty_hunter' else a.status_bounty_hunter end status_bounty_hunter,
+        case when a.status_funder is not null then 'funder' else a.status_funder end status_funder,
+        case when a.status_user is not null then 'user' else a.status_user end status_user,
+        case when b.bounty_owner_email is not null then 'funder' end status_funder_bounty
+    from
+        final a
+    left join
+        funder_restriction b
+    on
+        a.email = b.bounty_owner_email
+),
+user_join_dates as (
+    select distinct
+        b.id profile_id,
+        b.user_id,
+        b.handle,
+        coalesce(b.created_on, a.date_joined) join_date,  -- taking dashboard_profile join dates first as truth based on profile_id = 26
+        date_trunc('month', coalesce(b.created_on, a.date_joined)) join_month
+    from
+        dashboard_profile b
+    left join
+        auth_user a
+    on
+        b.user_id = a.id
+),
+result as (
+    select
+        a.profile_id,
+        a.email,
+        coalesce(a.status_funder_bounty, a.status_bounty_hunter, a.status_user) designation,
+        b.handle,
+        b.join_date
+    from
+        result_raw a
+    left join
+        user_join_dates b
+    on
+        a.profile_id = b.profile_id
+)
+select * from result;
+""".strip()
```

Comment:
did you generate this bit / write up the raw query yourself  ?  me curious

Reply:
As noted in the description, this query is one that Frank wrote up and uses on metabase, so I use it directly here. This allows us to adjust it easily rather than reconverting to ORM code each time we change it on metabase.